Source/EmbeddedBoundary/DistanceToEB.H

```diff
@@ -36,53 +36,90 @@ void normalize (amrex::RealVect& a) noexcept
 {
 #if (defined WARPX_DIM_3D)
     amrex::RealVect normal{0.0, 0.0, 0.0};
 
-    normal[0] -= phi(i,   j  , k  ) * dxi[0] * W[1][0] * W[2][0];
-    normal[0] += phi(i+1, j  , k  ) * dxi[0] * W[1][0] * W[2][0];
-    normal[0] -= phi(i,   j+1, k  ) * dxi[0] * W[1][1] * W[2][0];
-    normal[0] += phi(i+1, j+1, k  ) * dxi[0] * W[1][1] * W[2][0];
-    normal[0] -= phi(i,   j  , k+1) * dxi[0] * W[1][0] * W[2][1];
-    normal[0] += phi(i+1, j  , k+1) * dxi[0] * W[1][0] * W[2][1];
-    normal[0] -= phi(i  , j+1, k+1) * dxi[0] * W[1][1] * W[2][1];
-    normal[0] += phi(i+1, j+1, k+1) * dxi[0] * W[1][1] * W[2][1];
-
-    normal[1] -= phi(i,   j  , k  ) * dxi[1] * W[0][0] * W[2][0];
-    normal[1] += phi(i  , j+1, k  ) * dxi[1] * W[0][0] * W[2][0];
-    normal[1] -= phi(i+1, j  , k  ) * dxi[1] * W[0][1] * W[2][0];
-    normal[1] += phi(i+1, j+1, k  ) * dxi[1] * W[0][1] * W[2][0];
-    normal[1] -= phi(i,   j  , k+1) * dxi[1] * W[0][0] * W[2][1];
-    normal[1] += phi(i  , j+1, k+1) * dxi[1] * W[0][0] * W[2][1];
-    normal[1] -= phi(i+1, j  , k+1) * dxi[1] * W[0][1] * W[2][1];
-    normal[1] += phi(i+1, j+1, k+1) * dxi[1] * W[0][1] * W[2][1];
-
-    normal[2] -= phi(i  , j  , k  ) * dxi[2] * W[0][0] * W[1][0];
-    normal[2] += phi(i  , j  , k+1) * dxi[2] * W[0][0] * W[1][0];
-    normal[2] -= phi(i+1, j  , k  ) * dxi[2] * W[0][1] * W[1][0];
-    normal[2] += phi(i+1, j  , k+1) * dxi[2] * W[0][1] * W[1][0];
-    normal[2] -= phi(i,   j+1, k  ) * dxi[2] * W[0][0] * W[1][1];
-    normal[2] += phi(i  , j+1, k+1) * dxi[2] * W[0][0] * W[1][1];
-    normal[2] -= phi(i+1, j+1, k  ) * dxi[2] * W[0][1] * W[1][1];
-    normal[2] += phi(i+1, j+1, k+1) * dxi[2] * W[0][1] * W[1][1];
+    normal[0] -= phi(ic,   j  , k  ) * dxi[0] * Wc[0][0] * W[1][0] * W[2][0];
+    normal[0] += phi(ic+1, j  , k  ) * dxi[0] * Wc[0][0] * W[1][0] * W[2][0];
+    normal[0] -= phi(ic,   j+1, k  ) * dxi[0] * Wc[0][0] * W[1][1] * W[2][0];
+    normal[0] += phi(ic+1, j+1, k  ) * dxi[0] * Wc[0][0] * W[1][1] * W[2][0];
+    normal[0] -= phi(ic,   j  , k+1) * dxi[0] * Wc[0][0] * W[1][0] * W[2][1];
+    normal[0] += phi(ic+1, j  , k+1) * dxi[0] * Wc[0][0] * W[1][0] * W[2][1];
+    normal[0] -= phi(ic,   j+1, k+1) * dxi[0] * Wc[0][0] * W[1][1] * W[2][1];
+    normal[0] += phi(ic+1, j+1, k+1) * dxi[0] * Wc[0][0] * W[1][1] * W[2][1];
+    normal[0] -= phi(ic+1, j  , k  ) * dxi[0] * Wc[0][1] * W[1][0] * W[2][0];
+    normal[0] += phi(ic+2, j  , k  ) * dxi[0] * Wc[0][1] * W[1][0] * W[2][0];
+    normal[0] -= phi(ic+1, j+1, k  ) * dxi[0] * Wc[0][1] * W[1][1] * W[2][0];
+    normal[0] += phi(ic+2, j+1, k  ) * dxi[0] * Wc[0][1] * W[1][1] * W[2][0];
+    normal[0] -= phi(ic+1, j  , k+1) * dxi[0] * Wc[0][1] * W[1][0] * W[2][1];
+    normal[0] += phi(ic+2, j  , k+1) * dxi[0] * Wc[0][1] * W[1][0] * W[2][1];
+    normal[0] -= phi(ic+1, j+1, k+1) * dxi[0] * Wc[0][1] * W[1][1] * W[2][1];
+    normal[0] += phi(ic+2, j+1, k+1) * dxi[0] * Wc[0][1] * W[1][1] * W[2][1];
```

Comment:
I think this can be made more compact : 

```
for (int iic = 0; iic < 2; ++iic) {
    for (int kk = 0; kk < 2; ++kk) {
        for (int jj=0; jj< 2; ++jj) {
            for (int ii = 0; ii < 2; ++ii) {
                int icstart = ic + iic;
                amrex::Real sign = (i%2)*2. - 1.;
                int wccomp = static_cast<int>(icstart%2);
                int w1comp = static_cast<int>(jj%2);
                int w2comp = static_cast<int>(kk%2);
                normal[0] += sign * phi(icstart + ii, j + jj, k + kk) * dxi[0] * Wc[0][wccomp] * W[1][w1comp] * W[2][w2comp];
            }
        }
    }
}
```